feature/28_amenities

zilola08 opened the pull request:
Added amenities module, controller and service.

Zach13131 reviewed (approved):
approved

zilola08 commented:
Added swagger to amenities module

zilola08 commented:
Fixed merge conflicts

bakhromov96 commented on src/amenities/amenities.controller.ts:
```diff
@@ -0,0 +1,100 @@
+  @ApiOperation({ summary: 'GET AMENITIES LIST' })
+  @ApiUnauthorizedResponse()
+  @ApiResponse({
+    status: 201,
```
use correct status codes

zilola08 replied:
corrected the status codes

bakhromov96 commented on src/amenities/amenities.controller.ts:
```diff
@@ -0,0 +1,100 @@
+export class AmenitiesController {
+  constructor(private amenitiesService: AmenitiesService) {}
+
+  @Get('get/:id')
```
you can use just /:id

zilola08 replied:
corrected the endpoints

bakhromov96 commented on src/amenities/amenities.controller.ts:
```diff
@@ -0,0 +1,100 @@
+    return this.amenitiesService.getAmenities(id);
+  }
+
+  @Put('update/:id')
```
same here

zilola08 replied:
corrected the endpoints

bakhromov96 reviewed (changes requested):
I didn't put comment, because same comment apply for whole file.

zilola08 commented on prisma/migrations/20231204121925_update_amenities_table/migration.sql:
```diff
@@ -0,0 +1,30 @@
+-- AlterTable
+ALTER TABLE "Amenity" DROP COLUMN "numberOfRooms",
+DROP COLUMN "squareMeters",
+ADD COLUMN     "airConditioning" BOOLEAN NOT NULL DEFAULT false,
```
renamed all boolean columns as per example

zilola08 commented on prisma/schema.prisma:
```diff
@@ -62,11 +62,25 @@ model Accommodation {
-  wifi            Boolean
-  parking         Boolean
+  id               String  @id @default(uuid())
+  wifi             Boolean @default(false)
```
renamed all boolean columns

zilola08 commented:
made following changes as per comments:

- renamed all boolean columns;
- corrected status codes;
- corrected endpoints;
- corrected error messages and added custom errors;
- adjusted error-handling as per new global error filter;

Could you review the changes, please

gitkoshelew commented on src/amenities/amenities.service.ts:
```diff
@@ -0,0 +1,103 @@
+      return { message: 'Success updating amenities', data: updatedAmenities };
+    } catch (error) {
+      if (error instanceof PrismaClientKnownRequestError) {
+        if (error.code === 'P2025') {
```
add, please Enum for such numbers.  Or maybe it exists somewhere in prisma

mirsadikov replied:
Enums now created in another PR

Zach13131 replied:
Enums file already exists in develop branch

zilola08 replied:
added enums exclusively for prisma codes

bakhromov96 reviewed (approved):
Good job!
